src/openrct2/interface/Viewport.cpp

```diff
@@ -827,9 +827,33 @@ void viewport_render(
 #endif
 }
 
-static void viewport_fill_column(paint_session* session)
+static void record_session(const paint_session* session, std::vector<paint_session>* recorded_sessions, size_t record_index)
+{
+    // Perform a deep copy of the paint session, use relative offsets.
+    // This is done to extract the session for benchmark.
+    // Place the copied session at provided record_index, so the caller can decide which columns/paint sessions to copy; there
+    // is no column information embedded in the session itself.
+    if (recorded_sessions != nullptr)
```

Comment:
Makes more sense to check that before the call, also have you forgotten our coding style by now 😄

Reply:
Yeah, I figured as this is TU-local function I would rather put this check inside, since all our builds use LTO anyway and not clobber up the caller. I'll update it if that's what you prefer.

Tha clang-format check passes, do you have anything else in mind?

Reply:
Looks otherwise good. The reason why I want the check moved is that you can avoid the call and it makes it more clear.

Reply:
Updated